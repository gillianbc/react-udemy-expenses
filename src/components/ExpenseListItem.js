import React from "react";
import { Link } from "react-router-dom";

const ExpenseListItem = ({ id, amount, description, createdAt }) => {
  console.log("EXPENSE LIST ITEM", description);
  return (
    <div>
      <p>
        <Link to={"edit/" + id}>{description}</Link> Â£{amount} Created Date:{" "}
        {createdAt}.
      </p>
    </div>
  );
};

export default ExpenseListItem;
