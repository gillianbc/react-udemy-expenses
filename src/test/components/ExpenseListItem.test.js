import React from 'react'
import ExpenseListItem from '../../components/ExpenseListItem'
import { shallow } from 'enzyme'
import testExpenseData from '../resources/expenses'
import toJson from 'enzyme-to-json'
/*
* Tests the snapshot of the ExpenseListItem component
* */

// { id, amount, description, createdAt }
it('The snapshot should match when there is expense data', () => {
    // Remember, these curlies are not object brackets, they're JSX brackets
    const wrapper = shallow(<ExpenseListItem key={123} { ...testExpenseData[2]}/>);
    expect(toJson(wrapper)).toMatchSnapshot();
})


