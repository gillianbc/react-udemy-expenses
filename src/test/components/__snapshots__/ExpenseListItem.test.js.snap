// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The snapshot should match when there is expense data 1`] = `
<div>
  <p>
    <Link
      to="edit/3"
    >
      Credit Card
    </Link>
    Â£45.00
    Created Date: 
    10th February 1970
    .
  </p>
</div>
`;
